ci/docs-builder.py

```diff
@@ -0,0 +1,158 @@
+
+    # Get platform build version
+    repo_list = configuration['repositories']
+    try:
+        pulp_dict = filter(lambda x: x['name'] == 'pulp', repo_list)[0]
+    except IndexError:
+        raise RuntimeError("config file does not have an entry for 'pulp'")
+    version = pulp_dict['version']
+
+    if version.endswith('alpha'):
+        build_type = 'nightly'
+    elif version.endswith('beta'):
+        build_type = 'dev'
+    elif version.endswith('rc'):
+        build_type = 'dev'
+    else:
+        build_type = 'ga'
+
+    x_y_version = '.'.join(version.split('.')[:2])
+
+    builder.ensure_dir(WORKING_DIR, clean=True)
+
+    print "Getting git repos"
+    for component in get_components(configuration):
+        #clone the repos
+        branch_name = component['git_branch']
+        print "Cloning from github: %s" % component.get('git_url')
+        print "Switching to branch %s" % branch_name
+        clone_command = ['git', 'clone', component.get('git_url'), '--branch', branch_name]
+        exit_code = subprocess.call(clone_command, cwd=WORKING_DIR)
+        if exit_code != 0:
+            raise RuntimeError('An error occurred while cloning the repo.')
+
+    plugins_dir = WORKING_DIR + os.sep + 'pulp' + os.sep + 'docs' + os.sep + 'plugins'
+    builder.ensure_dir(plugins_dir)
+
+    for component in get_components(configuration):
+        if component['name'] == 'pulp':
+            continue
+
+        src = WORKING_DIR + os.sep + component['name'] + os.sep + 'docs'
+        dst = plugins_dir + os.sep + component['name']
+        os.symlink(src, dst)
+
+    # copy in the pulp_index.rst file
+    if x_y_version == '2.8' and build_type == 'ga':
+        src_path = 'docs/pulp_index_2_8.rst'
+    else:
+        src_path = 'docs/pulp_index.rst'
+    pulp_index_rst = WORKING_DIR + os.sep + 'pulp' + os.sep + 'docs' + os.sep + 'index.rst'
+    copyfile(src_path, pulp_index_rst)
+
+    # copy in the plugin_index.rst file
+    plugin_index_rst = plugins_dir + os.sep + 'index.rst'
+    copyfile('docs/plugin_index.rst', plugin_index_rst)
+
+    # copy in the all_content_index.rst file
+    all_content_index_rst = WORKING_DIR + os.sep + 'pulp' + os.sep + 'docs' + os.sep + 'all_content_index.rst'
+    copyfile('docs/all_content_index.rst', all_content_index_rst)
+
+    # build the docs via the Pulp project itself
+    print "Building the docs"
+    docs_directory = WORKING_DIR + os.sep + 'pulp' + os.sep + 'docs'
+    make_command = ['make', 'html', 'SPHINXOPTS=-Wn']
+    exit_code = subprocess.call(make_command, cwd=docs_directory)
+    if exit_code != 0:
+        raise RuntimeError('An error occurred while building the docs.')
+
+    # rsync the docs to the root if it's GA of latest
+    if build_type == 'ga' and x_y_version == LATEST:
+        local_path_arg = docs_directory + os.sep + '_build' + os.sep + 'html' + os.sep
+        remote_path_arg = '57600fb10c1e664383000229@docs-pulp.rhcloud.com:~/app-root/repo/diy/'
```

Comment:
This hostname appears multiple times in the script and looks like something that might change in the future; generating paths to the rsync remote should probably be generalized, and the hostname for the rsync remote should probably be a top-level constant, or otherwise called out as a configuration value.